runtime/assembly/modules/storage.ts

```diff
@@ -36,4 +38,48 @@ export namespace Storage {
         const value64: u64 = Serialiser.serialiseResultwOutRetain(value);
         ext_storage_set_version_1(key64, value64);
     }
+    /**
+     * Gets the given key from storage, placing the value into a buffer and returning the number of bytes that the
+     * entry in storage has beyond the offset. 
+     * ArrayBuffer has fixed length of 20.
+     * Returns Option instance with populated ArrayBuffer concatenated to the result from storage_read
+     * @param key key of the data to get
+     * @param offset offset beyond which the value is copied
+     */
+    export function read(key: u8[], offset: i32): Option<u8[]> {
+        const key64: u64 = Serialiser.serialiseResultwOutRetain(key);
+        const valueOut = new ArrayBuffer(20);
+        let valuePtr = changetype<usize>(valueOut);
+        let valueSize = valueOut.byteLength;
+        const valueOut64 = ((valueSize as u64) << 32) | valuePtr;
```

Comment:
Maybe it will be useful to create a new function in the `Serialiser` for `serialiseBufferwOutRetain`.